sys/posix/pthread/include/pthread_once.h

```diff
@@ -24,14 +24,17 @@ extern "C" {
 /**
  * @brief           Datatype to supply to pthread_once().
  */
-typedef volatile int pthread_once_t;
+typedef struct {
+  int   is_initialized;  /**< initialized */
+  int   init_executed;   /**< init function executed */
+} pthread_once_t;
 
 /**
  * @def             PTHREAD_ONCE_INIT
  * @brief           Initialization for pthread_once_t.
  * @details         A zeroed out pthread_once_t is initialized.
```

Comment:
Is this still true?

Reply:
What do mean?

Reply:
Ah ok, you mean the comment of PTHREAD_ONCE_INIT. Hm, it not completely zeroed out. I will change it.
May I squash it directly to avoid an additional round in CI?

Reply:
What would` {0, 0}` represent?

Reply:
> What would` {0, 0}` represent?

It would mean that the variable isn't initalized using `PTHREAD_ONCE_INIT`. We could add an 
```
assert(once_control->is_initialized)
```
but it wouldn't have any consequence for the execution of the `init_routine`. So it doesn't matter whether it is really initialized before. `is_initialized` member is just for compatibility for `pthread_once_t` as required by GCC.

Reply:
Ok so the comment *does* need an update - because you can't just `memset(&once_control, 0, sizeof(once_control))` or `static pthread_once_t once_control;` anymore as the comment suggests.

Reply:
> because you can't just `memset(&once_control, 0, sizeof(once_control))`

You could do it because the `is_initialized` member isn't used at all. For the execution of the `init_routine` it doesn't make a difference whether it is set to 1 or 0.

By the way, the documentation is already updated, isn't that enough?